src/adapter/zboss/frame.ts

```diff
@@ -103,20 +106,40 @@ export function readZBOSSFrame(buffer: Buffer): ZBOSSFrame {
     const buf = new ZBOSSBuffaloZcl(buffer);
     const version = buf.readUInt8();
     const type = buf.readUInt8();
-    const commandId = buf.readUInt16();
+    const commandId: CommandId = buf.readUInt16();
     let tsn = 0;
+
     if ([FrameType.REQUEST, FrameType.RESPONSE].includes(type)) {
         tsn = buf.readUInt8();
     }
-    const payload = readPayload(type, commandId, buf);
 
-    return {
-        version,
-        type,
-        commandId,
-        tsn,
-        payload,
-    };
+    const zdoResponseClusterId = ZBOSS_COMMAND_ID_TO_ZDO_RSP_CLUSTER_ID[commandId];
+
+    if (zdoResponseClusterId !== undefined) {
+        const category = buf.readUInt8(); // XXX: should always be ZDO here?
+        const zdoClusterId = zdoResponseClusterId;
+        const zdo = BuffaloZdo.readResponse(false, zdoResponseClusterId, buffer.subarray(6));
+
+        return {
+            version,
+            type,
+            commandId,
+            tsn,
+            payload: {
+                category,
+                zdoClusterId,
```

Comment:
need status attribute for success controll:
`status: zdo[0],`

Reply:
Success control will be done upstream (Controller/Device, etc), `Zdo.Buffalo` handles reading based on status and passes it back to caller (will allow workaround for some quirky devices in the future).

So, removing this chunk:
https://github.com/Nerivec/zigbee-herdsman/blob/6ba120509c2cee2c3415e7fdbf72964da56cfa5f/src/adapter/zboss/driver.ts#L357-L359
should fix it?

Reply:
if the status is controlled by the caller, then Yes, there is no need for driver-level control for ZDO commands

Reply:
Last two commits should take care of the two issues. Can you confirm?

Reply:
it is better, but there is a problem - when connecting the device, a frame of a slightly different format comes:

```
debug:  zh:zboss:uart: <-- FRAME: 00020c020cda603602602bd5b3708e
debug:    zh:zboss:uart: <-- error RangeError: The value of "offset" is out of range. It must be >= 0 and <= 8. Received 10
    at new NodeError (node:internal/errors:406:5)
    at boundsError (node:internal/buffer:88:9)
    at Buffer.readUInt8 (node:internal/buffer:254:5)
    at BuffaloZdo.readUInt8 (/home/kirov/Projects/zigbee2mqtt/node_modules/zigbee-herdsman/src/buffalo/buffalo.ts:34:35)
    at BuffaloZdo.readEndDeviceAnnounce (/home/kirov/Projects/zigbee2mqtt/node_modules/zigbee-herdsman/src/zspec/zdo/buffaloZdo.ts:1815:35)
    at Function.readResponse (/home/kirov/Projects/zigbee2mqtt/node_modules/zigbee-herdsman/src/zspec/zdo/buffaloZdo.ts:1517:32)
    at readZBOSSFrame (/home/kirov/Projects/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/zboss/frame.ts:152:32)
    at ZBOSSUart.onPackage (/home/kirov/Projects/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/zboss/uart.ts:265:41)

```

Reply:
I'm looking for a way to fix it

Reply:
```
   /**
     * Request: [transaction sequence number: 1]
     *          [node ID:2] [EUI64:8] [capabilities:1]
     */
    END_DEVICE_ANNOUNCE = 0x0013,
```
```     // Device announce indication
    // [CommandId.ZDO_DEV_ANNCE_IND]: {
    //     request: [],
    //     response: [],
    //     indication: [
    //         {name: 'nwk', type: DataType.UINT16},
    //         {name: 'ieee', type: DataType.IEEE_ADDR},
    //         {name: 'macCapabilities', type: DataType.UINT8},
    //     ],
    // },
```
The problem here is that a frame with the FrameType.INDICATION type arrives, which means there is no category in the frame, but data immediately goes.
Therefore, for this type of frame, you need to skip the line:
`const category = buf.readUInt8();`

Reply:
and something else...

Reply:
Darn, missed that one! Should be fixed.